docs/madr/decisions/006-mesh-traffic-permission.md

```diff
@@ -0,0 +1,576 @@
+with "ALLOW" action and with a single "MeshTrafficPermission" policy. The policy has permission "any: true" and the list of "principals".
+List of principals generated based on the list from step 3. Each principal is matched with OR semantic: 
+
+```yaml
+action: ALLOW
+policies:
+  "MeshTrafficPermission":
+    permissions:
+      - any: true
+    principals: 
+      - and_ids:
+          ids:
+            - authenticated:
+                principal_name:
+                  exact: "kuma://kuma.io/zone/us-east"
+            - not_id:
+                authenticated:
+                  principal_name:
+                    exact: "kuma://env/dev"
+```
+
+Similarly generate a v3.RBAC using "shadow_action: ALLOW". 
+Generated RBACs are set in the network filter as "rules" and "shadow_rules" respectively.
+
+4. If MeshTrafficPermission is targeting proxy then NetworkRBAC is placed as a NetworkFilter on the inbound listener.
+If MeshTrafficPermission is targeting MeshHTTPRoute then NetworkRBAC is placed as a NetworkFilter on the route. 
+Even in the case of MeshHTTPRoute, global per-proxy NetworkRBAC is still will be set to provide secure access for other routes. 
+
+### MeshTrafficPermission and Kubernetes
+
+MeshTrafficPermission is going to be the first policy with a new policy matching. 
+There are a few things worth mentioning regarding this fact. 
+
+#### Mesh as label
+
+Instead of creating a CRD with a "mesh" field, we can use a label to store mesh:
+
+```yaml
+apiVersion: kuma.io/v1alpha1
+kind: MeshTrafficPermission
+metadata:
+  labels:
+    kuma.io/mesh: mesh-1
+```
+
+This allows filtering policies by mesh using kubectl. 
+If the label is unset it will be added automatically by the webhook to be "kuma.io/mesh: default".
+
+#### Scope is namespaced
+
+Today all policies are cluster-scoped, but new policies should be namespace scoped.
+MeshTrafficPermission can only be created in the namespace where the control-plane runs (`kuma-system` by default), 
+and affects DPPs across all namespaces.
+In the future we'll add support for other namespaces, so the policy could affect DPPs only in the specified namespace but this is out of scope for this MADR. 
+
+### Cross-mesh
+
+MeshTrafficPermission is used to specify a list of clients from other meshes that have an access to MeshGatewayRoute.
+In that case we use targetRef with "mesh" field:
+
+```yaml
+type: MeshTrafficPermission
+mesh: expose
+name: cross-mesh-permission
+spec:
+  targetRef:
+    kind: MeshGatewayRoute
+    name: cross-mesh-route
+  from:
+    - targetRef:
+        kind: MeshService
+        name: backend
+        mesh: consume
+      default:
+        action: ALLOW
+    - targetRef:
+        kind: MeshServiceSubset
+        name: backend
+        mesh: consume
+        tags:
+          version: v1
+      default:
+        action: DENY
```

Comment:
So how will this work with all your negations btw? I guess it work there's just a "special" match for `targetRef: kind Mesh name: consume` where the principal matching is prefixed based right?

Reply:
When we generate a rule-based view with negations, we can add an artificial `kuma.io/mesh` tag. So we should look at 2 cases: `kuma.io/mesh: consume` and `kuma.io/mesh: !consume`